### PR TITLE
[NCL-2100] refactored tests for json output

### PR DESCRIPTION
**Pull request by stanlyDoge**
~~…pendencyRelationship~~

### CODE REVIEW

**Comment by stanlyDoge** [comment]
retest this please

**Comment by janinko** [comment]
retest this please

**Comment by janinko** [comment]
retest this please

File: communication/src/main/java/org/jboss/da/communication/aprox/model/GAVDependencyTree.java [hunk 1]
```diff
@@ -100,4 +100,12 @@ private static void pruneLeaf(GAVDependencyTree tree,
...
         return this.gav.compareTo(o.gav);
     }
+
+    public int size() {
+        int result = 1;
+        for (GAVDependencyTree dependency : dependencies) {
+            result += dependency.size();
+        }
+        return result;
+    }
```
Comment (janinko) [inline communication/src/main/java/org/jboss/da/communication/aprox/model/GAVDependencyTree.java]: This isn't used, do we want to keep it?

File: communication/src/main/java/org/jboss/da/communication/pom/impl/DependencyTreeBuilder.java [hunk 2]
```diff
@@ -158,6 +160,14 @@ public boolean equals(Object obj) {
...
             return true;
         }
+
+        public int size() {
+            int tmp = 1;
+            for (PathNode child : childs) {
+                tmp += child.size();
+            }
+            return tmp;
+        }
```
Comment (janinko) [inline communication/src/main/java/org/jboss/da/communication/pom/impl/DependencyTreeBuilder.java]: This isn't used, do we want to keep it?

File: communication/src/main/java/org/jboss/da/communication/pom/impl/DependencyTreeBuilder.java [hunk 3]
```diff
@@ -166,14 +176,26 @@ public boolean equals(Object obj) {
...
-            GAV targetGav2 = GalleyWrapper.generateGAV(target2);
 
-            int ret = targetGav1.compareTo(targetGav2);
-            return ret;
+            int ret = target1.getGroupId().compareTo(target2.getGroupId());
+            if (ret == 0) {
+                ret = target1.getArtifactId().compareTo(target2.getArtifactId());
+            }
+            if (ret == 0) {
+                ret = target1.getVersionString().compareTo(target2.getVersionString());
```
Comment (janinko) [inline communication/src/main/java/org/jboss/da/communication/pom/impl/DependencyTreeBuilder.java]: Use VersionComparator

File: reports-rest/src/main/java/org/jboss/da/rest/reports/Reports.java [hunk 4]
```diff
@@ -100,7 +100,7 @@ public Response scmGenerator(@ApiParam(value = "scm information") SCMLocator scm
         try {
 
             Optional<ArtifactReport> artifactReport = reportsGenerator.getReportFromSCM(scm);
-
+
```
Comment (janinko) [inline reports-rest/src/main/java/org/jboss/da/rest/reports/Reports.java]: ?

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by janinko** [comment]
retest this please

**Comment by janinko** [comment]
:+1:  (if the test pass)

**Comment by thescouser89** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by jbartece** [comment]
retest this please

**Comment by janinko** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by janinko** [comment]
Please split the fixes and test changes into 2 separate PRs

**Comment by stanlyDoge** [comment]
retest this please

**Comment by janinko** [comment]
https://github.com/project-ncl/dependency-analysis/pull/460

**Pull request closed without merging**